packages/website/blog/2023-07-09-announcing-typescript-eslint-v6.md

```diff
@@ -629,19 +629,23 @@ import * as TSESLint from '@typescript-eslint/utils/dist/ts-eslint';
 
 That presents a few issues for developers:
 
-- It can be unclear which of many potential import paths to use
-- TypeScript sometimes suggests importing types or values meant to be private
-- Consumers using deep import paths can be broken by internal refactors that rename files
+- It can be unclear which of many potential import paths to use.
+- TypeScript sometimes suggests importing types or values meant to be private.
+- Consumers using deep import paths can be broken by internal refactors that rename files.
 
 As of [feat: add package.json exports for public packages](https://github.com/typescript-eslint/typescript-eslint/pull/6458), `@typescript-eslint/*` packages now use `exports` to prevent importing internal file paths.
 Developers must now mostly import directly from the package names, e.g.:
 
 ```ts
-import * as TSESLint from '@typescript-eslint/utils/ts-eslint';
+import * as TSESLint from '@typescript-eslint/utils';
```

Comment:
Checking with @bradzacher as previous editor of this line - what is the right import we'd to convey here?

Reply:
This has already been fixed in main:
https://github.com/typescript-eslint/typescript-eslint/blob/6addee9b4ef668cd2e026130083a3a783b327e9e/packages/website/blog/2023-07-09-announcing-typescript-eslint-v6.md#L636-L641

Reply:
Shouldn't we recommend this instead?

```ts
import { TSESLint } from '@typescript-eslint/utils'; 
```

IMO star imports are not great.

Reply:
It depends - you'll find a lot of people we're actually doing things like this before:

```ts
import { RuleModule } from '@typescript-eslint/utils/dist/ts-eslint' ;
```

So for those usecases switching to

```ts
import { RuleModule } from '@typescript-eslint/utils/ts-eslint' ;
```

Is the better migration!

However if someone was using the `TSESLint` namespace before then it's unlikely that they were deeply importing it - so they wouldn't need to update anything.

If we were to do anything to update this here - we'd probably want to update the examples to be "realer".

Reply:
> we'd probably want to update the examples to be "realer".

Ok, I updated the PR to say this:

```ts
// import * as TSESLint from '@typescript-eslint/utils/dist/ts-eslint';
// -->
import { TSESLint } from '@typescript-eslint/utils';

// import { RuleModule } from '@typescript-eslint/utils/dist/ts-eslint';
// -->
import { RuleModule } from '@typescript-eslint/utils/ts-eslint';

// import { AST_NODE_TYPES } from "@typescript-eslint/types/dist/generated/ast-spec";
// -->
import { AST_NODE_TYPES } from "@typescript-eslint/types";
```

Reply:
cc @bradzacher?